### PR TITLE
Expanding vanilla prices

### PR DESCRIPTION
**Pull request by esztermarton**
Editing vanilla prices to take in cost_of_carries and discount_rates and continuous dividend while keeping compatibility with version that takes in discount_factors.

### CODE REVIEW

**Comment by esztermarton** [comment]
@googlebot I signed it!

File: tf_quant_finance/black_scholes/vanilla_prices.py [hunk 1]
```diff
@@ -64,21 +68,41 @@ def option_price(*,
     forwards: A real `Tensor` of any shape that broadcasts to the shape of
       `volatilities`. The forwards to maturity. Either this argument or the
       `spots` must be supplied but both must not be supplied.
+    discount_rates: An optional real `Tensor` of same dtype as the
+      `volatilities`. If not `None`, discount factors are calculated as e^(-rT),
```
Comment (cyrilchim) [inline tf_quant_finance/black_scholes/vanilla_prices.py]: Please add the shape specification. Like this

An optional `Tensor` of same dtype as the `volatilities` and of the shape that broadcasts with `volatilities`

The same for dividends and cost of carries

Comment (esztermarton) [reply tf_quant_finance/black_scholes/vanilla_prices.py]: done!

**Comment by cyrilchim** [comment]
We have now pushed it! Thank you

**Pull request merged**